docs/tools.md

```diff
@@ -529,6 +529,100 @@ print(test_model.last_model_request_parameters.function_tools)
 
 _(This example is complete, it can be run "as is")_
 
+## Agent-wide Dynamic Tool Preparation {#prepare-tools}
+
+In addition to per-tool `prepare` methods, you can also define an agent-wide `prepare_tools` function. This function is called at each step and allows you to filter or modify the list of all tool definitions available to the agent for that step. This is especially useful if you want to enable or disable multiple tools at once, or apply global logic based on the current context.
```

Comment:
What does "at each step" means? 🤔 Isn't it just called before sending it to the provider?

Reply:
I think I meant to say it almost the same way as the prepare method is described [above](https://github.com/pydantic/pydantic-ai/pull/1474/files#diff-1f2a047633300b387d1d53d25e910a8759221a3492eee35231306bc30cb12740R430)... Would it be okay to change it to...

```suggestion
In addition to per-tool `prepare` methods, you can also define an agent-wide `prepare_tools` function. This function is called at each step of a run and allows you to filter or modify the list of all tool definitions available to the agent for that step. This is especially useful if you want to enable or disable multiple tools at once, or apply global logic based on the current context.
```